server/routes/connections.js

```diff
@@ -2,11 +2,13 @@ import _ from 'lodash';
 import { Router } from 'express';
 
 import multipartRequest from '../lib/multipartRequest';
+// only fetch one page of connections
+const CONNECTIONS_FETCH_LIMIT = 50;
```

Comment:
This limit should be equal or slightly more than `export const CONNECTIONS_LIST_LIMIT = 20000;`.

Explanation:
`CONNECTIONS_LIST_LIMIT` is the UI limit to control the logic between text and list. However this logic will be executed based on the list returned by backend.

`CONNECTIONS_FETCH_LIMIT` is the backend limit to control at most how many items will be retrieved. If more than this number, we'll cap at this number and report back to frontend. 

For `CONNECTIONS_LIST_LIMIT` based frontend logic to work properly, we need to report cap `CONNECTIONS_FETCH_LIMIT` at lest at `CONNECTIONS_FETCH_LIMIT`, if possible slightly higher.

Right now, the connections will always be capped at 50 regardless of how many are there and everyone will see fist 50 connections in a list.

Note: As an alternative, we can fetch the `count`, notice that it exceeds the limit and skip traversing altogether and tell frontend "too much to fetch" in a special response. This would have the same effect with significantly less calls.